docroot/themes/contrib/civictheme/tests/src/Unit/CivicThemeIsExternalLinkUnitTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Drupal\Tests\civictheme\Unit;
+
+/**
+ * Class CivicThemeIsExternalLinkUnitTest.
+ *
+ * Test cases for _civictheme_link_is_external().
+ *
+ * @group CivicTheme
+ */
+class CivicThemeIsExternalLinkUnitTest extends CivicThemeUnitTestBase {
+
+  /**
+   * Test for civictheme_link_is_external().
+   *
+   * @dataProvider dataProviderIsExternalLink
+   */
+  public function testParse($url, $host, $overridden_domains, $expected) {
+    $actual = civictheme_link_is_external($url, $host, $overridden_domains);
+
+    $this->assertEquals($expected, $actual);
+  }
+
+  /**
+   * Data provider for testParse().
+   */
+  public function dataProviderIsExternalLink() {
+    $host = 'http://test.com';
+    $overridden_domains = [
+      'http://overridden.com',
+      'http://overriddendomain.com',
+    ];
+    return [
+      // Link without domain.
```

Comment:
please always start with empty values - function should be able to handle those correctly

Reply:
please do not use variables in tests - tests should be explicit as much as possible - please use actual values